### PR TITLE
Implement Template System

### PR DESCRIPTION
**Pull request by MrinallU**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

- [x] I have tested my code
- [x] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [x] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

**Review by RealMyth21 — APPROVED** [review]
looks good

**Comment by RealMyth21** [comment]
just keep in mind, it failed build

**Comment by MrinallU** [comment]
I'll fix that.

File: src/components/Groups/GroupPage/FeedItem.tsx [hunk 1]
```diff
@@ -77,11 +92,116 @@ export default function FeedItem({
...
+            getFirestore(firebaseApp),
+            'groups',
+            key,
+            'posts',
+            docRef.id,
+            'problems'
+          )
+        );
+
+        problems.map(async problem => {
```
Comment (maggie-j-liu) [inline src/components/Groups/GroupPage/FeedItem.tsx:172]: problems are not exported... maybe you need to await this (also wrap in `Promise.all`?)

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]
is there a way to export multiple posts at once? can you make a way to copy an entire group?

File: src/components/Groups/GroupPage/FeedItem.tsx (hunk 1 above)
Comment (MrinallU) [reply src/components/Groups/GroupPage/FeedItem.tsx:172]: Forgot to add that, I'll fix it.

**Comment by MrinallU** [comment]
> is there a way to export multiple posts at once? can you make a way to copy an entire group?

So for the first question: yeah I can do that pretty easily. But as for copying an entire group, would it just be the same group wiht a different name?

**Comment by maggie-j-liu** [comment]
> would it just be the same group with a different name?

basically, but it would allow the person who copied to become owner of the group and remove all other members (so we could make a "template" group for clubs, then club leaders can copy the entire group instead of having to set everything up themselves

**Pull request closed without merging**